keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DiffFingerprintRepository.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.persistence
+
+import com.netflix.spinnaker.keel.api.ResourceId
+import com.netflix.spinnaker.keel.diff.ResourceDiff
+import java.security.MessageDigest
+import javax.xml.bind.DatatypeConverter
+
+/**
+ * Stores a hash of the diff
+ */
+interface DiffFingerprintRepository {
```

Comment:
Is there a reason this isn't a part of `ResourceRepository`? It is entirely `resourceId` based.

Reply:
Discussed offline - for now, we're going to keep this separate because it's tracking info about the resource vs core resource things.